### ISSUE TITLE
Command Line Option Not Supported In All Versions Of Gcc

### ISSUE THREAD

**Issue by OxyDeadbeef** [alexcrichton/gcc-rs#43]
I get this error:
`
c++: error: unrecognized command line option ‘-stdlib=libstdc++’
`
When I try to compile c++ code from my rust build script with .cpp(true)
It appears as though `-stdlib=libstdc++` is not an accepted parameter by my version of gcc.

**Comment by alexcrichton** [comment]
cc @TimNN, upon re-reading #39 I'm thinking the best option is to just not pass `-stdlib` by default (and instead rely on the compiler default). You mentioned that this can cause link errors on OSX though, so I'm wondering if you can elaborate on that?

**Comment by TimNN** [comment]
The problem was that `gcc-rs` was unconditionally instructing cargo to link the rust code to `libstdc++` when compiling with `.ccp(true)`. [Original code](https://github.com/alexcrichton/gcc-rs/blob/b8da3f50a0a0b03e7fa0103e9dc8be9f776b9ce3/src/lib.rs#L178-L181).

On OS X the default compiler is clang, which however links to `libc++` thus resulting in linking errors.

As such the [`cpp_stdlib`](http://alexcrichton.com/gcc-rs/gcc/struct.Config.html#method.cpp_stdlib) option was added. It allows choosing which stdlib to use and if one is choosen, it will force the compiler to use this library (using the `-stdlib=` flag) and instructs cargo to link the rust code to this library.

@OxyDeadbeef which OS / compiler version are you using?

For the moment you can work around the problem by manually creating a [`Config`](http://alexcrichton.com/gcc-rs/gcc/struct.Config.html) and setting `config.cpp_stdlib(None)`. You will then have to manually link to `libstdc++` by writing `println!("cargo:rustc-link-lib=stdc++");` in your `build.rs` file.

**Comment by TimNN** [comment]
Alright I did some digging and might have been a little careless when testing my original pull request:

OS X provides a `g++` in `/usr/bin`, which is just a wrapper around `clang` and thus supports `-stdlib=` and I have installed a 'real' `g++` in `/usr/local/bin` (which I usually have before `/usr/bin` in my path). For some reason while testing the code I have was using the OS X provided `g++` and thus everything worked.

Proposed solution: Change `cpp_stdlib` to not automatically specify the `-stdlib=` flag (but keep detecting the stdlib to use based on the OS and instructing cargo to link to it) and add another option (or modify `cpp_stdlib` to take a second parameter) along the lines of `force_cpp_stdlib_usage`, which, if set, would pass the `-stdlib` flag to the compiler.
